sources/subs/InlinePermissionsForm.class.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/**
+ * Functions to support the permissions controller
+ *
+ * @name      ElkArte Forum
+ * @copyright ElkArte Forum contributors
+ * @license   BSD http://opensource.org/licenses/BSD-3-Clause
+ *
+ * @version 1.1 beta 1
+ */
+
+/**
+ * Class to initialize inline permissions sub-form and save its settings
+ */
+class Inline_Permissions_Form
+{
+	/**
+	 * Save the permissions of a form containing inline permissions.
+	 *
+	 * @param string[] $permissions
+	 */
+	public static function save($permissions)
+	{
+		global $context;
+
+		$db = database();
+
+		// No permissions? Not a great deal to do here.
+		if (!allowedTo('manage_permissions'))
+			return;
+
+		// Almighty session check, verify our ways.
+		checkSession();
+		validateToken('admin-mp');
+
+		// Make sure they can't do certain things,
+		// unless they have the right permissions.
+		Permissions::loadIllegal();
+
+		$insertRows = array();
+		foreach ($permissions as $permission)
+		{
+			if (!isset($_POST[$permission]))
+				continue;
+
+			foreach ($_POST[$permission] as $id_group => $value)
+			{
+				if (in_array($value, array('on', 'deny')) && (empty($context['illegal_permissions']) || !in_array($permission, $context['illegal_permissions'])))
+					$insertRows[] = array((int) $id_group, $permission, $value == 'on' ? 1 : 0);
+			}
+		}
+
+		// Remove the old permissions...
+		$db->query('', '
+			DELETE FROM {db_prefix}permissions
+			WHERE permission IN ({array_string:permissions})
+			' . (empty($context['illegal_permissions']) ? '' : ' AND permission NOT IN ({array_string:illegal_permissions})'),
+			array(
+				'illegal_permissions' => !empty($context['illegal_permissions']) ? $context['illegal_permissions'] : array(),
+				'permissions' => $permissions,
+			)
+		);
+
+		// ...and replace them with new ones.
+		if (!empty($insertRows))
+			$db->insert('insert',
+				'{db_prefix}permissions',
+				array('id_group' => 'int', 'permission' => 'string', 'add_deny' => 'int'),
+				$insertRows,
+				array('id_group', 'permission')
+			);
```

Comment:
Good point, this should do https://github.com/elkarte/Elkarte/pull/2663/commits/91fd9737ede039399a234602fbe2e4631c9a6dd1 (I hope)